### PR TITLE
Change add application credential flow

### PR DESCRIPTION
**Pull request by bramkragten**
## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/translations/en.json [hunk 1]
```diff
@@ -3060,6 +3058,7 @@
           "editor": {
             "caption": "Add Credential",
             "description": "OAuth is used to grant Home Assistant access to information on other websites without giving a passwords. This mechanism is used by companies such as Spotify, Google, Withings, Microsoft, and Twitter.",
+            "missing_credentials": "Setting up {integration} requires configuring application credentials.",
```
Comment (balloob) [inline src/translations/en.json:3061]: Can we make sure we link to the domain documentation of the integration with instructions? 
```suggestion
            "missing_credentials": "Setting up {integration} requires configuring application credentials. [See the documentation for more information.]({url})",
```

**Review by balloob — DISMISSED** [review]

**Comment by balloob** [comment]
Current look:

image


CC @allenporter

File: src/panels/config/application_credentials/dialog-add-application-credential.ts [hunk 2]
```diff
@@ -107,19 +110,31 @@ export class DialogAddApplicationCredential extends LitElement {
...
-                      this.hass.localize,
-                      this._domain!
-                    ),
+                    integration: selectedDomainName,
                   }
                 )}
+                <a
+                  href=${documentationUrl(
+                    this.hass,
+                    `/integrations/${this._domain}`
```
Comment (bramkragten) [inline src/panels/config/application_credentials/dialog-add-application-credential.ts:119]: Should we account for custom integrations here?

**Comment by allenporter** [comment]
Is this removing the integration specific documentation?

Screen Shot 2022-10-02 at 10 32 21 AM

Would be helpful to have some context in the PR description to understand the goal, and what exactly is changing given the existing is complicated with a lot of steps.

File: src/panels/config/application_credentials/dialog-add-application-credential.ts (hunk 2 above)
Comment (allenporter) [reply src/panels/config/application_credentials/dialog-add-application-credential.ts:119]: Is there a way to grab the integration documentation url from the manifest?

**Comment by allenporter** [comment]
> Is this removing the integration specific documentation?
> 
> Screen Shot 2022-10-02 at 10 32 21 AM
> 
> Would be helpful to have some context in the PR description to understand the goal, and what exactly is changing given the existing is complicated with a lot of steps.

OK, it appears to be there for Nest and Sheets when I test locally.  Calendar works on my prod instance for me, but not locally. Maybe a separate issue with my test instance, but it also was not there in your sceenshot either.

I see we're able to get rid of some of the dialog steps, that's great. Where there other changes to the config flow or did I just miss how to do this the first time?

**Comment by balloob** [comment]
Yeah my screenshot is bad. There was a bug that dropped integration specific instructions. We're going to drop the intro to application credentials and just show the integration specific intro.

**Comment by balloob** [comment]
image

**Comment by bramkragten** [comment]
> I see we're able to get rid of some of the dialog steps, that's great. Where there other changes to the config flow or did I just miss how to do this the first time?

There where other changes, but not related to this I think. We just simplified the flow for the user now

**Comment by bramkragten** [comment]
> OK, it appears to be there for Nest and Sheets when I test locally. Calendar works on my prod instance for me, but not locally. Maybe a separate issue with my test instance, but it also was not there in your sceenshot either.

We only loaded translations for integrations that where loaded, so setting up a new integration would not load the translations

**Comment by allenporter** [comment]
> > OK, it appears to be there for Nest and Sheets when I test locally. Calendar works on my prod instance for me, but not locally. Maybe a separate issue with my test instance, but it also was not there in your sceenshot either.
> 
> We only loaded translations for integrations that where loaded, so setting up a new integration would not load the translations

Ah that would explain it. Is this something fixable by one of us?

Thanks for all the cleanup here.

**Comment by bramkragten** [comment]
> > > OK, it appears to be there for Nest and Sheets when I test locally. Calendar works on my prod instance for me, but not locally. Maybe a separate issue with my test instance, but it also was not there in your sceenshot either.
> > 
> > 
> > We only loaded translations for integrations that where loaded, so setting up a new integration would not load the translations
> 
> Ah that would explain it. Is this something fixable by one of us?
> 
> Thanks for all the cleanup here.

It is fixed by this PR :-)

**Pull request merged**